tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs

```diff
@@ -24,12 +25,12 @@ internal class CIAgentWriter : IEventWriter
         private static Span[] _spanArray;
         private readonly AgentWriter _agentWriter;
 
-        public CIAgentWriter(ImmutableTracerSettings settings, ISampler sampler, int maxBufferSize = DefaultMaxBufferSize)
+        public CIAgentWriter(ImmutableTracerSettings settings, ISampler sampler, IDiscoveryService discoveryService = null, int maxBufferSize = DefaultMaxBufferSize)
```

Comment:
I'd argue this _shouldn't_ default to `null` (or be able to be null), as `StatsAggregator` will throw if it is. Alternatively `StatsAggregator` should explicitly allow and handle it (by adding `#nullable enable` etc

Reply:
Yeah you're right, I think I just got a little lazy here 😞 I've updated this to remove the default value in commit 4a765a2520f1b35918e0e420433f9d2f3492c4a1